specs/experimental/gov-delegation.md

```diff
@@ -309,17 +305,28 @@ enum AllowanceType {
 | `Absolute`  | `0`    | The amount of votes delegated is fixed and denominated in the `GovernanceToken`'s decimals  |
 | `Relative`  | `1`    | The amount of votes delegated is relative and denominated in percentages.                   |
 
-## Backwards Compatibility
+## Migration
 
-The `GovernanceDelegation` contract ensures backwards compatibility by allowing the migration of delegation state from the
-`GovernanceToken`.
+All write functions in the `GovernanceDelegation` MUST check if the users interacting with it (such as creating or updating
+a delegation) have been migrated by checking the `migrated` mapping from its [storage](#storage). If a user has not been
+migrated, the `GovernanceDelegation` MUST copy the delegation and checkpoint data from the token contract to its own state.
+After copying the data, the `GovernanceDelegation` MUST update the `migrated` mapping to reflect that the address has been
+migrated, and clear the delegation in the `GovernanceToken` contract.
 
-## Migration
+The `GovernanceDelegation` MUST enforce the following invariants for the migration logic:
+
+1. A user MUST NOT have an active delegation in both `GovernanceToken` and `GovernanceDelegation` at the same time.
```

Comment:
Technically some of these statements are implied by the other statements but I think it's good enough to just be explicit and list all of the statements

Reply:
yeah i think it helps to have some redundancy in this specific case